pkg/clients/ecclient/ecclient.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package ecclient
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/eestream"
+	"storj.io/storj/pkg/ranger"
+	proto "storj.io/storj/protos/overlay"
+)
+
+// ECClient defines an interface for storing erasure coded data to piece store nodes
+type ECClient interface {
+	Put(ctx context.Context, nodes []proto.Node, es eestream.ErasureScheme,
+		pieceID PieceID, data io.Reader, expiration time.Time) error
+	Get(ctx context.Context, nodes []proto.Node, es eestream.ErasureScheme,
+		pieceID PieceID, size int64) (ranger.Ranger, error)
+	Delete(ctx context.Context, nodes []proto.Node, pieceID PieceID) error
+}
+
+type ecClient struct {
+	t TransportClient
+}
+
+// New creates an ECClient
+func New(t TransportClient) ECClient {
+	return &ecClient{t: t}
+}
+
+func (ec *ecClient) Put(ctx context.Context, nodes []proto.Node, es eestream.ErasureScheme,
+	pieceID PieceID, data io.Reader, expiration time.Time) error {
+	// TODO config/param for minimum and optimum thresholds and max buffer memory?
```

Comment:
okay hmm. i think there's two things for the minimum/optimum thresholds - we could either add them to ErasureScheme, or make a new "RedundancyStrategy" type that contains an ErasureScheme and those values. what do you think? 

for the max buffer memory, let's make that be a constructor argument?